harness/determined/cli/user.py

```diff
@@ -218,6 +243,110 @@ def edit(args: argparse.Namespace) -> None:
+    if data:
+        json_data = [elem.to_json() for elem in data]
+        if len(data) == 1:
+            json_data = json_data[0]
+    else:
+        json_data = {}
+    if args.yaml:
+        print(util.yaml_safe_dump(json_data, default_flow_style=False))
+    else:
+        render.print_json(json_data)
+
+
+def describe_token(args: argparse.Namespace) -> None:
+    sess = cli.setup_session(args)
+    result = []
+    for arg in args.username:
+        userID = bindings.get_GetUserByUsername(session=sess, username=arg).user.id
+        resp = bindings.get_GetAccessToken(session=sess, userId=userID)
+        result.append(resp.tokenInfo)
+    render_token_info(result)
+
+
+def list_tokens(args: argparse.Namespace) -> None:
+    sess = cli.setup_session(args)
+    resp = bindings.get_GetAllAccessTokens(sess, includeInactive=args.all)
+    if args.json or args.yaml:
+        print_token_info(resp.tokenInfo, args)
+        return
+    render_token_info(resp.tokenInfo)
+
+
+def revoke_token(args: argparse.Namespace) -> None:
+    sess = cli.setup_session(args)
+    try:
+        request = bindings.v1PatchAccessTokenRequest(
+            tokenId=args.token_id, description=None, setRevoked=True
+        )
+        resp = bindings.patch_PatchAccessToken(sess, body=request, tokenId=args.token_id)
+        print(json.dumps(resp.to_json(), indent=2))
+    except api.errors.NotFoundException:
+        raise errors.CliError("Token not found")
+    print("Successfully updated token with ID: {}".format(args.token_id))
+
+
+def create_token(args: argparse.Namespace) -> None:
+    sess = cli.setup_session(args)
+    current_user, token_user = sess.username, args.username
+    request, handler = None, None
+    if token_user is None or token_user == current_user:
+        user_id = bindings.get_GetUserByUsername(session=sess, username=current_user).user.id
+    else:
+        user_id = bindings.get_GetUserByUsername(session=sess, username=token_user).user.id
+    request = bindings.v1PostAccessTokenRequest(
+        lifespan=args.lifespan, userId=user_id, description=args.description
+    )
+    handler = functools.partial(bindings.post_PostAccessToken, userId=user_id)
+    resp = handler(session=sess, body=request).to_json()
+
+    outputString = None
+    if args.yaml:
+        outputString = util.yaml_safe_dump(resp, default_flow_style=False)
+    elif args.json:
+        outputString = json.dumps(resp, indent=2)
+    else:
+        outputString = resp["token"]
+
+    if args.file_path:
+        with open(args.file_path, "w") as file:
+            file.write(outputString)
+    else:
+        print(outputString)
+
+
+def update_token(args: argparse.Namespace) -> None:
+    sess = cli.setup_session(args)
+    try:
+        if args.description and args.token_id:
+            request = bindings.v1PatchAccessTokenRequest(
+                tokenId=args.token_id, description=args.description, setRevoked=False
```

Comment:
can this accidentally un-revoke a token?  That would feel like maybe not a security bug, but like a security "surprise".

Reply:
user will not be able to Patch any already revoked tokens. They would receive that error before Patch takes place.

Reply:
Resolving it as it is now implemented